src/viking.js

```diff
@@ -1,11 +1,120 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack() {
+    return this.strength;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  //Viking has access to all of "Soldier"'s methods, including "attack()" via "extends"
+
+  //Damage
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health >= 1) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  //BattleCry RAAAAAAA!
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
-// Saxon
-class Saxon {}
+//Saxon
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health >= 1) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else {
+      return `A Saxon has died in combat`;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+
+  vikingAttack() {
+    let rViking = Math.floor(Math.random() * this.vikingArmy.length);
+    let rSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    let attack = this.vikingArmy[rViking].strength;
+    this.saxonArmy[rSaxon].receiveDamage(attack);
```

Comment:
save the return of the call to the receiveDamage function in a variable and then return the value of that variable after the splice.
this way you can also get rid of line 79! :)